README.md

```diff
@@ -375,96 +378,131 @@ switch (anInput) {
 
 ## Annotations
 
-Standard annotations should be used - in particular `@Override`. This should
-appear the line before the function declaration.
+Standard annotations should be used - in particular `override`. This should appear on the same line as the function declaration.
 
 __BAD:__
 
-```java
-protected void onCreate(Bundle savedInstanceState) {
+```kotlin
+fun onCreate(savedInstanceState: Bundle?) {
   super.onCreate(savedInstanceState);
 }
 ```
 
 __GOOD:__
 
-```java
-@Override
-protected void onCreate(Bundle savedInstanceState) {
+```kotlin
+override fun onCreate(savedInstanceState: Bundle?) {
   super.onCreate(savedInstanceState);
 }
 ```
 
+## Types 
 
-## XML Guidance
-
-Since Android uses XML extensively in addition to Java, we have some rules
-specific to XML.
+Always use Kotlin's native types when available. Kotlin is JVM-compatible so **[TODO: more info]**
 
-### XML File Names
+### Type Inference
 
-View-based XML files should be prefixed with the type of view that they
-represent.
+Type inference should be preferred where possible to explicitly declared types. 
 
 __BAD:__
 
-- `login.xml`
-- `main_screen.xml`
-- `rounded_edges_button.xml`
+```kotlin
+val something: MyType = MyType()
+val meaningOfLife: Int = 42
+```
 
 __GOOD:__
 
-- `activity_login.xml`
-- `fragment_main_screen.xml`
-- `button_rounded_edges.xml`
+```kotlin
+val something = MyType()
+val meaningOfLife = 42
+```
 
-### Indentation
+### Constants vs. Variables 
+
+Constants are defined using the `val` keyword, and variables with the `var` keyword. Always use `val` instead of `var` if the value of the variable will not change.
+
+*Tip*: A good technique is to define everything using `val` and only change it to `var` if the compiler complains!
+
+### Companion Objects
+
+** TODO: A bunch of stuff about companion objects **
+
+### Optionals
+
+Declare variables and function return types as optional with `?` where a `null` value is acceptable.
+
+Use implicitly unwrapped types declared with `!!` only for instance variables that you know will be initialized before use, such as subviews that will be set up in `onCreate` for an Activity or `onCreateView` for a Fragment.
```

Comment:
I think we probably need a whole section on `lateinit` - maybe you can cover that in the next PR? :trollface: